docarray/helper.py

```diff
@@ -321,3 +325,66 @@ def dataclass_from_dict(klass, dikt):
         if isinstance(dikt, (tuple, list)):
             return [dataclass_from_dict(klass.__args__[0], f) for f in dikt]
         return dikt
+
+
+def protocol_and_compress_from_file_path(
+    file_path: str,
+    default_protocol: Optional[str] = None,
+    default_compress: Optional[str] = None,
+) -> Tuple[Optional[str], Optional[str]]:
+    """Extract protocol and compression algorithm from a string, use defaults if not found.
+
+    :param file_path: path of a file.
+    :param default_protocol: default serialization protocol used in case not found.
+    :param default_compress: default compression method used in case not found.
+
+    Examples:
+
+    >>> protocol_and_compress_from_file_path('./docarray_fashion_mnist.protobuf.gzip')
+    ('protobuf', 'gzip')
+
+    >>> protocol_and_compress_from_file_path('/Documents/docarray_fashion_mnist.protobuf')
+    ('protobuf', None)
+
+    >>> protocol_and_compress_from_file_path('/Documents/docarray_fashion_mnist.gzip')
+    (None, gzip)
+    """
+
+    protocol = default_protocol
+    compress = default_compress
+
+    file_extensions = [e.replace('.', '') for e in pathlib.Path(file_path).suffixes]
+    for extension in file_extensions:
+        if extension in ALLOWED_PROTOCOLS:
+            protocol = extension
+        elif extension in ALLOWED_COMPRESSIONS:
+            compress = extension
```

Comment:
```suggestion
    m = re.match('^.*\.([^.]+)\.([^.]+)$', file_path)
    if m and m.groups() == 2 and m.group(1) in ALLOWED_PROTOCOLS and m.group(2) in ALLOWED_COMPRESSIONS:
        protocol, compress = m.group(1), m.group(2)
```

Reply:
You can decide no compression. I'm allowing the order not to be relevant but the values that is why I did it how I did.